src/shared/bus-util.c

```diff
@@ -1060,13 +1060,13 @@ static int map_basic(sd_bus *bus, const char *member, sd_bus_message *m, sd_bus_
 
         case SD_BUS_TYPE_BOOLEAN: {
                 unsigned b;
-                int *p = userdata;
+                bool *p = userdata;
```

Comment:
uh, this one is problematic, i am not sure we should do that. Currently, we never expose the bool type in our public APIs, as we avoid too many C99isms in that. We only use it internally. That means the whole of sd-bus uses "int" for booleans in its public API (though "bool" internally wherever applicable). Now, src/shared/bus-util.c is not really part of sd-bus, and is internally, but the already existing pitfalls of using different types for booleans is increased I think if bus-util.c doesn't follow sd-bus in this regard.

I also have the suspicion that there might be various users already that expect bools to be ints which will trip up on this type change. 

I am pretty sure we should find a different solutoin for this, for example, by providing an explicit setter function for bools here, i.e. add bus_map_bool() similar to bus_map_id128(), which does the necessary conversion to bool.

Reply:
Hmm. I see. But I do not want to specify `bus_map_bool()` to so many properties. I've tried to support this problem in another way in #8546. Please take a look.